packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,782 @@
+import { Base64 } from 'js-base64';
+import { first, flow, partial, result, trim } from 'lodash';
+import {
+  localForage,
+  APIError,
+  ApiRequest,
+  unsentRequest,
+  responseParser,
+  Entry,
+  AssetProxy,
+  PersistOptions,
+  readFile,
+  CMS_BRANCH_PREFIX,
+  generateContentKey,
+  parseContentKey,
+  labelToStatus,
+  isCMSLabel,
+  EditorialWorkflowError,
+  statusToLabel,
+} from 'netlify-cms-lib-util';
+
+export const API_NAME = 'Azure DevOps';
+
+export class AzureRepo {
+  org: string;
+  project: string;
+  name: string;
+
+  constructor(location?: string | null) {
+    if (!location || location == undefined) {
+      throw new Error(
+        "An Azure repository must be specified in the format 'organisation/project/repo'.",
+      );
+    }
+
+    const components = trim(location, '/').split('/', 3);
+    this.org = components[0];
+    this.project = components[1];
+    this.name = components[2] || components[1];
+  }
+}
+
+export interface AzureUser {
```

Comment:
I like all these type definitions 😄

Reply:
Out of interest, would you consider it good practise to do class-per-file? I considered creating a few folders as modules and organising like that, but didn't want to break with what looked like wider convention. I'm usually a .NET dev, old habits die hard : )

Reply:
I'm up for separating classes to different files once its needed. Doesn't seems like the case here as they are only used in one place.